zipline/assets/asset_writer.py

```diff
@@ -81,237 +83,176 @@ class AssetDBWriter(with_metaclass(ABCMeta)):
-            CREATE TABLE IF NOT EXISTS futures_exchanges (
-            exchange_id INTEGER NOT NULL,
-            exchange TEXT,
-            timezone TEXT
-        )""")
-
-        c.execute("""
-            CREATE TABLE IF NOT EXISTS futures_root_symbols (
-            root_symbol_id INTEGER NOT NULL,
-            root_symbol TEXT,
-            sector TEXT,
-            description TEXT,
-            exchange_id INTEGER{fk}
-        )""".format(fk=", FOREIGN KEY(exchange_id) REFERENCES "
-                    "futures_exchanges(exchange_id)"
-                    if constraints else ""))
-
-        c.execute("""
-            CREATE TABLE IF NOT EXISTS futures_contracts (
-            sid INTEGER NOT NULL,
-            symbol TEXT,
-            root_symbol_id INTEGER,
-            root_symbol TEXT,
-            asset_name TEXT,
-            start_date INTEGER DEFAULT 0,
-            end_date INTEGER,
-            first_traded INTEGER,
-            exchange_id INTEGER,
-            exchange TEXT,
-            notice_date INTEGER,
-            expiration_date INTEGER,
-            contract_multiplier REAL{fk}
-        )""".format(fk=", FOREIGN KEY(exchange_id) REFERENCES "
-                    "futures_exchanges(exchange_id), "
-                    "FOREIGN KEY(root_symbol_id) REFERENCES "
-                    "futures_root_symbols(root_symbol_id)"
-                    if constraints else ""))
-
-        c.execute("""
-            CREATE TABLE IF NOT EXISTS asset_router
-            (sid integer,
-            asset_type text
-        )""")
-
-        # Note: Would be optimal to use INTEGER PRIMARY KEY here, but SQLite
-        # tables cannot be modified after creation. Using CREATE UNIQUE INDEX
-        # only marginally less performant.
-        if constraints:
-
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_equities_sid '
-                      'ON equities(sid)')
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_equities_symbol '
-                      'ON equities(symbol)')
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_equities_fuzzy '
-                      'ON equities(fuzzy)')
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_futures_exchanges_en '   # noqa
-                      'ON futures_exchanges(exchange_id)')
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_futures_contracts_sid '  # noqa
-                      'ON futures_contracts(sid)')
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_futures_root_symbols_id '  # noqa
-                      'ON futures_root_symbols(root_symbol_id)')
-            c.execute('CREATE UNIQUE INDEX IF NOT EXISTS ix_asset_router_sid  '
-                      'ON asset_router(sid)')
-
-        # Note: Also need a max_date table.
-
-        db_conn.commit()
+        self.sql_metadata = metadata = sa.MetaData(bind=engine)
+        self.equities = sa.Table(
+            'equities',
+            metadata,
+            sa.Column(
```

Comment:
This will be unique if ```primary_key=True```, which in turns will depend on ```constraints=True```. We should change the default of ```constraints``` to be ```True``` so that it will be compatible with FFC.